node/network/gossip-support/src/lib.rs

```diff
@@ -150,23 +163,30 @@ impl State {
 		for leaf in leaves {
 			let current_index = util::request_session_index_for_child(leaf, ctx.sender()).await.await??;
 			let maybe_new_session = match self.last_session_index {
-				Some(i) if i >= current_index => None,
+				Some(i) if current_index <= i && !self.force_request => None,
 				_ => Some((current_index, leaf)),
 			};
 
 			if let Some((new_session, relay_parent)) = maybe_new_session {
 				tracing::debug!(target: LOG_TARGET, %new_session, "New session detected");
 				let authorities = determine_relevant_authorities(ctx, relay_parent).await?;
 				ensure_i_am_an_authority(keystore, &authorities).await?;
-				tracing::debug!(target: LOG_TARGET, num = ?authorities.len(), "Issuing a connection request");
+				let num = authorities.len();
+				tracing::debug!(target: LOG_TARGET, %num, "Issuing a connection request");
 
-				connect_to_authorities(
+				let failures = connect_to_authorities(
 					ctx,
 					authorities,
 					PeerSet::Validation,
 				).await;
 
+				// we await for the request to be processed
+				// this if fine, it should take much less time than one session
+				let failures = failures.await.unwrap_or(0);
```

Comment:
```suggestion
				let failures = failures.await.unwrap_or(num);
```

This is safer right?

Reply:
yes!

Reply:
actually, it doesn't matter, if the future returns an error, that means network bridge is shutting down (as is the node)